### ISSUE TITLE
Error: Failed to load latest transactions from YNAB Caused by: YNAB API request error Caused by: unknown variant ``, expected one of `red`, `orange`, `yellow`, `green`, `blue`, `purple` at line 1 column 169

### ISSUE THREAD

**Issue by wizonesolutions** [borsboom/foreign-currency-accounts-for-ynab#17]
fca4ynab is currently stuck after I unflagged a few transactions. I had green flags on them before, and then I removed them. I specifically used Toolkit for YNAB's "Remove Flags" context menu feature. Toolkit for YNAB is a client-side tool, and I don't think it should be able to put YNAB into an inconsistent state, but I'm not sure. Maybe I should try re-flagging them, running the tool, and unflagging them by hand again or something?

But I'm hoping it can just catch the error and handle it somehow.

Error message:

```
 !  ~/S/fca4ynab  env RUST_LOG="fca4ynab=debug" ./fca4ynab --yes --auto-approve-transactions=true --auto-approve-adjustments=true
[2019-12-15T20:59:38Z DEBUG fca4ynab::cli] Using configuration file path: "/Users/kevin/Library/Preferences/io.borsboom.fca4ynab/env"
[2019-12-15T20:59:38Z DEBUG fca4ynab::database] Using database file: /Users/kevin/Library/Application Support/io.borsboom.fca4ynab/data.sqlite3
Loading latest transactions from YNAB...
Error: Failed to load latest transactions from YNAB
Caused by: YNAB API request error
Caused by: unknown variant ``, expected one of `red`, `orange`, `yellow`, `green`, `blue`, `purple` at line 1 column 169
```

**Comment by borsboom** [comment]
Thanks for the detailed report.  I've been able to reproduce this, but only using Toolkit for YNAB.  Removing flags using the standard (non-toolkit) UI seems to work fine.

According to the YNAB API's Swagger spec, an empty string for the flag colour field is illegal.  The flag field should either be `null` or one of the flag colour values.  My guess is that the Toolkit is directly setting to to the empty string in the browser's memory, and then when that gets synced the server isn't actually checking it carefully enough (probably using a dynamically typed programming language) and just putting it into their content store as-is.

I'll special-case the parser (which is auto-generated from the Swagger spec) to handle an empty string here.

In the mean-time, you can work around this problem by setting a flag again and then using the standard UI to clear them (one at a time, unfortunately).

**Comment by wizonesolutions** [comment]
Thanks. I'll file a report with them too. It was only three flags or so, so
shouldn't be too bad.

**Comment by borsboom** [comment]
I've pushed PR #18 to handle this.  Once the artifacts have finished building, you can try it out by [downloading the artifacts](https://dev.azure.com/borsboom/foreign-currency-accounts-for-ynab/_build/results?buildId=197&view=artifacts&type=publishedArtifacts).

**Comment by borsboom** [comment]
Released in [v0.1.9](https://github.com/borsboom/foreign-currency-accounts-for-ynab/releases/tag/release/0.1.9).
